src/api_client.rs

```diff
@@ -1,560 +1,596 @@
-    pub finalized: Checkpoint,
-}
-
-#[derive(Serialize, Deserialize)]
-pub enum ValidatorStatus {
-    PendingInitialized,
-    PendingQueued,
-    ActiveOngoing,
-    ActiveExiting,
-    ActiveSlashed,
-    ExitedUnslashed,
-    ExitedSlashed,
-    WithdrawalPossible,
-    WithdrawalDone,
-    // TODO what are these?
-    Active,
-    Pending,
-    Exited,
-    Withdrawal,
-}
-
-pub enum PubkeyOrIndex {
-    Pubkey(BlsPublicKey),
-    Index(ValidatorIndex),
-}
-
-pub struct ValidatorDescriptor {
-    pub pubkey_or_index: PubkeyOrIndex,
-    pub status: ValidatorStatus,
-}
-
-#[derive(Serialize, Deserialize)]
-pub struct ValidatorSummary {
-    #[serde(with = "crate::serde::as_string")]
-    pub index: ValidatorIndex,
-    #[serde(with = "crate::serde::as_string")]
-    pub balance: Gwei,
-    pub status: ValidatorStatus,
-    pub validator: Validator,
-}
-
-#[derive(Serialize, Deserialize)]
-pub struct BalanceSummary {
-    #[serde(with = "crate::serde::as_string")]
-    pub index: ValidatorIndex,
-    #[serde(with = "crate::serde::as_string")]
-    pub balance: Gwei,
-}
-
-pub struct CommitteeFilter {
-    pub epoch: Option<Epoch>,
-    pub index: Option<CommitteeIndex>,
-    pub slot: Option<Slot>,
-}
-
-#[derive(Serialize, Deserialize)]
-pub struct CommitteeSummary {
-    #[serde(with = "crate::serde::as_string")]
-    pub index: CommitteeIndex,
-    #[serde(with = "crate::serde::as_string")]
-    pub slot: Slot,
-    #[serde(with = "crate::serde::collection_over_string")]
-    pub validators: Vec<ValidatorIndex>,
-}
-
-#[derive(Serialize, Deserialize)]
-pub struct SyncCommitteeSummary {
-    #[serde(with = "crate::serde::collection_over_string")]
-    pub validators: Vec<ValidatorIndex>,
-    // TODO fix serde here
-    pub validator_aggregates: Vec<Vec<ValidatorIndex>>,
-}
-
-#[derive(Serialize, Deserialize)]
-pub struct BeaconHeaderSummary {
-    pub root: Root,
-    pub canonical: bool,
-    pub signed_header: SignedBeaconBlockHeader,
-}
-
-pub enum EventTopic {
-    Head,
-    Block,
-    Attestation,
-    VoluntaryExit,
-    FinalizedCheckpoint,
-    ChainReorg,
-    ContributionAndProof,
-}
-
-#[derive(Serialize, Deserialize)]
-pub struct NetworkIdentity {
-    pub peer_id: PeerId,
-    pub enr: Enr,
-    pub p2p_addresses: Vec<Multiaddr>,
-    pub discovery_addresses: Vec<Multiaddr>,
-    pub metadata: MetaData,
-}
-
-#[derive(Serialize, Deserialize)]
-pub enum PeerState {
-    Disconnected,
-    Connecting,
-    Connected,
-    Disconnecting,
-}
-
-#[derive(Serialize, Deserialize)]
-pub enum ConnectionOrientation {
-    Inbound,
-    Outbound,
-}
-
-#[derive(Serialize, Deserialize)]
-pub struct PeerDescriptor {
-    pub state: PeerState,
-    pub direction: ConnectionOrientation,
-}
-
-#[derive(Serialize, Deserialize)]
-pub struct PeerDescription {
-    pub peer_id: PeerId,
-    pub enr: Enr,
-    pub last_seen_p2p_address: Multiaddr,
-    pub state: PeerState,
-    pub direction: ConnectionOrientation,
-}
+#[tokio::main]
```

Comment:
so this attribute `#[tokio::main]` should only go around the "main" entrypoint

you added one in `src/main.rs` so lets move this there.